thrust/testing/unittest/testframework.h

```diff
@@ -284,7 +284,46 @@ enum TestStatus
 using ArgumentSet = std::set<std::string>;
 using ArgumentMap = std::map<std::string, std::string>;
 
-std::vector<size_t> get_test_sizes();
+// clang-format off
+constexpr size_t standard_test_sizes[] =
+{
+  0, 1, 2, 3, 4, 5, 8, 10, 13, 16, 17, 19, 27, 30, 31, 32,
+  33, 35, 42, 53, 58, 63, 64, 65, 72, 97, 100, 127, 128, 129, 142, 183, 192, 201, 240, 255, 256,
+  257, 302, 511, 512, 513, 687, 900, 1023, 1024, 1025, 1565, 1786, 1973, 2047, 2048, 2049, 3050, 4095, 4096,
+  4097, 5030, 7791, 10000, 10027, 12345, 16384, 17354, 26255, 32768, 43718, 65533, 65536,
+  65539, 123456, 131072, 731588, 1048575, 1048576,
+  3398570, 9760840, (1 << 24) - 1, (1 << 24),
+  (1 << 24) + 1, (1 << 25) - 1, (1 << 25), (1 << 25) + 1, (1 << 26) - 1, 1 << 26,
+  (1 << 26) + 1, (1 << 27) - 1, (1 << 27)
+};
+// clang-format on
+
+constexpr size_t tiny_threshold    = 1 << 5; //   32
+constexpr size_t small_threshold   = 1 << 8; //  256
+constexpr size_t medium_threshold  = 1 << 12; //   4K
+constexpr size_t default_threshold = 1 << 16; //  64K
+constexpr size_t large_threshold   = 1 << 20; //   1M
+constexpr size_t huge_threshold    = 1 << 24; //  16M
+constexpr size_t epic_threshold    = 1 << 26; //  64M
+constexpr size_t max_threshold     = (std::numeric_limits<size_t>::max)();
+
+inline std::vector<size_t> test_sizes = [] {
+  std::vector<size_t> v;
+  for (size_t s : standard_test_sizes)
+  {
+    if (s <= default_threshold)
+    {
+      v.push_back(s);
+    }
+  }
+  return v;
+}();
+
+inline std::vector<size_t> get_test_sizes()
```

Comment:
Question: This is copying the test_sizes vector every time, should it return a reference?

Reply:
Yes